lib/mqtt_msg.c

```diff
@@ -272,6 +272,17 @@ char *mqtt_get_publish_data(uint8_t *buffer, size_t *length)
     return (char *)(buffer + i);
 }
 
+char *mqtt_get_suback_data(uint8_t *buffer, size_t *length)
+{
+    // SUBACK payload length = total length - (fixed header (2 bytes) + variable header (2 bytes))
+    if ((int)*length - 4 > 0) {
```

Comment:
Nitpick: maybe easier without a cast
```suggestion
    if (*length > 4) {
```

Reply:
OMG.  Facepalm. I'll change that.

Reply:
I added a comment about the "remaining length"-length which needs to be 1. This also commented further down in the code.